File: C/121_maxProfit.c
int maxProfit(int* prices, int pricesSize) {
    int min_price = prices[0];
    int max_profit = 0;
    int i;
    int temp;

    for (i = 0; i < pricesSize; i++) {
        if (prices[i] < min_price) 
            min_price = prices[i];
        
        if ((temp = prices[i] - min_price) > max_profit)
            max_profit = temp;
    }
    return max_profit;
}

File: python/121_maxProfit.py
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) < 2:
            return 0
            
        low_profit = prices[0]
        max_profit = 0
        for p in prices:
            low_profit = min(low_profit, p)
            max_profit = max(max_profit, p - low_profit)
        
        return max_profit
